The error on the Windows build is:
```
2024-12-28T01:16:41.1617961Z   image_io.cpp
2024-12-28T01:16:41.6068343Z D:\a\qaseprite\qaseprite\aseprite\third_party\zlib\zlib.h(34,10): error C1083: Cannot open include file: 'zconf.h': No such file or directory [D:\a\qaseprite\qaseprite\build\aseprite\src\doc\doc-lib.vcxproj]
2024-12-28T01:16:41.6069671Z   (compiling source file '../../../../aseprite/src/doc/image_io.cpp')
```
Which might be related to these earlier messages:
```
2024-12-28T01:12:13.9906174Z -- Renaming
2024-12-28T01:12:13.9906587Z --     D:/a/qaseprite/qaseprite/aseprite/third_party/zlib/zconf.h
2024-12-28T01:12:13.9907222Z -- to 'zconf.h.included' because this file is included with zlib
2024-12-28T01:12:13.9907809Z -- but CMake generates it automatically in the build directory.
```
Seems like either CMake didn't generate the `zconf.h` after all, or the location at which it was generated wasn't added to the include path.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -107,11 +107,17 @@ jobs:
         include:
         - qt_version: 5.15.2
           qt_version_major: 5
-          qt_arch: win64_msvc2019_64
+          qt_arch: win32_mingw81
+          mingw_component: tools_mingw
+          mingw_variant: qt.tools.win32_mingw810
+          mingw_dir: mingw810_32
           arch: 32
         - qt_version: 6.8.1
           qt_version_major: 6
-          qt_arch: win64_msvc2022_64
+          qt_arch: win64_mingw
+          mingw_component: tools_mingw1310
+          mingw_variant: qt.tools.win64_mingw1310
+          mingw_dir: mingw1310_64
           arch: 64
 
     defaults:
@@ -134,10 +140,11 @@ jobs:
       with:
         version: ${{ matrix.qt_version }}
         arch: ${{ matrix.qt_arch }}
+        tools: "${{ matrix.mingw_component }},${{ matrix.mingw_variant }}"
         cache: true
 
     - name: Build
       run: |
-        cmake -B build -DCMAKE_BUILD_TYPE=Release 
+        cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=i686-w64-mingw32-gcc.exe -DCMAKE_CXX_COMPILER=i686-w64-mingw32-g++.exe
         cmake --build build --config Release
-        cmake --install build --config Release 
+        cmake --install build --config Release

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -109,16 +109,10 @@ jobs:
           qt_version_major: 5
           qt_arch: win64_msvc2019_64
           arch: 32
-          openssl_arch: x86
-          filename_suffix: 'Windows-7-8_x86'
-          qt_modules: ''
         - qt_version: 6.8.1
           qt_version_major: 6
           qt_arch: win64_msvc2022_64
-          qt_modules: ''
           arch: 64
-          openssl_arch: x64
-          filename_suffix: 'Windows-10+_x86_64'
 
     defaults:
       run:
@@ -140,9 +134,6 @@ jobs:
       with:
         version: ${{ matrix.qt_version }}
         arch: ${{ matrix.qt_arch }}
-        modules: ${{ matrix.qt_modules }}
-        aqtversion: '==3.1.19'
-        setup-python: false
         cache: true
 
     - name: Build

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -144,7 +144,11 @@ jobs:
         cache: true
 
     - name: Build
+      env:
+        CC: i686-w64-mingw32-gcc.exe
+        CXX: i686-w64-mingw32-g++.exe
       run: |
-        cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=i686-w64-mingw32-gcc.exe -DCMAKE_CXX_COMPILER=i686-w64-mingw32-g++.exe
+        cmake -B build -DCMAKE_BUILD_TYPE=Release \
+          -DUSE_SHARED_ZLIB=on
         cmake --build build --config Release
         cmake --install build --config Release

